packages/twenty-server/src/engine/metadata-modules/field-metadata/services/field-metadata.service-v2.ts

```diff
@@ -188,6 +198,78 @@ export class FieldMetadataServiceV2 {
     });
   }
 
+  async updateOne({
+    updateFieldInput,
+    workspaceId,
+  }: {
+    updateFieldInput: UpdateFieldInput;
+    workspaceId: string;
+  }) {
+    const { flatObjectMetadataMaps: existingFlatObjectMetadataMaps } =
+      await this.workspaceMetadataCacheService.getExistingOrRecomputeFlatObjectMetadataMaps(
+        { workspaceId },
+      );
+
+    const inputTranspilationResult =
+      fromUpdateFieldInputToFlatFieldMetadataToUpdate({
+        existingFlatObjectMetadataMaps,
+        updateFieldInput,
+      });
+
+    if (inputTranspilationResult.status === 'fail') {
+      throw inputTranspilationResult.error;
+    }
+
+    const updatedFlatFieldMetadata = inputTranspilationResult.result;
+
+    const validationErrors =
+      await this.flatFieldMetadataValidatorService.validateFlatFieldMetadataUpdate(
+        {
+          existingFlatObjectMetadataMaps,
+          flatFieldMetadataToValidate: updatedFlatFieldMetadata,
+          workspaceId,
+        },
+      );
+
+    if (validationErrors.length > 0) {
+      throw new MultipleMetadataValidationErrors(
+        validationErrors,
+        'Multiple validation errors occurred while creating field',
+      );
+    }
+
+    try {
+      const fromFlatObjectMetadataMaps = getSubFlatObjectMetadataMapsOrThrow({
+        flatObjectMetadataMaps: existingFlatObjectMetadataMaps,
+        objectMetadataIds: [updatedFlatFieldMetadata.objectMetadataId],
+      });
+      const toFlatObjectMetadataMaps =
+        replaceFlatFieldMetadataInFlatObjectMetadataMapsOrThrow({
+          flatObjectMetadataMaps: fromFlatObjectMetadataMaps,
+          flatFieldMetadata: updatedFlatFieldMetadata,
+        });
+      const workspaceMigration = this.workspaceMigrationBuilderV2.build({
+        fromFlatObjectMetadataMaps,
+        toFlatObjectMetadataMaps,
+        inferDeletionFromMissingObjectFieldIndex: false,
+        workspaceId,
+      });
+
+      await this.workspaceMigrationRunnerV2Service.run(workspaceMigration);
+
+      return this.fieldMetadataRepository.findOne({
```

Comment:
nit: findOne might return undefined
```suggestion
      return this.fieldMetadataRepository.findOneOrFail({
```
this should be caught below in the case the record was not properly created . Up to you!